MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/ClickTrace.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.macrobenchmark.target
+
+import android.os.Looper
+import android.view.MotionEvent
+import androidx.tracing.Trace
+import curtains.*
+
+/**
+ * Logs async trace sections that track the duration of click handling in the app, i.e. the
+ * duration from MotionEvent.ACTION_UP to OnClick callbacks.
+ */
+object ClickTrace {
+
+    private const val SECTION_NAME = "ClickTrace"
+
+    private var clickInProgress = false
+
+    private val isMainThread: Boolean get() = Looper.getMainLooper().thread === Thread.currentThread()
+
+    fun onClickPerformed() {
+        Trace.endAsyncSection(SECTION_NAME, 0)
+        checkMainThread()
+        check(clickInProgress) {
+            "No click in progress"
+        }
+        clickInProgress = false
+    }
+
+    fun install() {
+        checkMainThread()
+        Curtains.onRootViewsChangedListeners += OnRootViewAddedListener { rootView ->
+            rootView.phoneWindow?.let { window ->
+                window.touchEventInterceptors += OnTouchEventListener { event ->
+                    if (event.action == MotionEvent.ACTION_UP) {
+                        Trace.beginAsyncSection(SECTION_NAME, 0)
```

Comment:
This is very cool. 👏🏽 

Using async traces this way gives us a way to capture this metric in a manner that is very generalizable.